### ISSUE TITLE
Allow explicit importing of dynamically imported submodules

### ISSUE THREAD

**Issue by shukob** [isl-org/Open3D#3146]
**Is your feature request related to a problem? Please describe.**
I took some time to figure out I cannot import like
```python
from open3d.camera import PinholeCameraIntrinsic
```
because `open3d.camera` is actually `open3d.cpu.pybind.camera`.
It is currently included as just `camera` by
```
            from open3d.cuda.pybind import (camera, geometry, io, pipelines,
                                            utility, t)
```
in `open3d/__init__.py`.

**Describe the solution you'd like**
To be consistent with the document that says the module is `open3d.camera`, we can additionally define submodules by reflection like
```python
setattr(open3d, 'camera', getattr(open3d.cpu.pybind, 'camera'))
```
. Overall
```python
for submodule in ['camera', 'geometry', 'io', 'pipelines', 'utility', 't']:
    setattr(open3d, submodule, getattr(open3d.cpu.pybind, submodule))
```
. The same can be done for gpu.

**Describe alternatives you've considered**

**Additional context**

**Comment by yxlao** [comment]
@shukob thanks! That's an interesting idea. Looks like we only need to modify `python/open3d/__init__.py` to achieve this. 

Would you mind creating a pull request for this? We'll be happy to review and test it.

**Comment by ssheorey** [comment]
Resolved by PR #5741
